nexus-tweaks.user.js

```diff
@@ -32,6 +32,391 @@ function timesCharExist(x, c){
   return (x.match(new RegExp(c,'g')) || []).length;
 }
 
+function truncate(input, maxSize) {
+  if (input.length > maxSize) {
+    return input.substring(0, maxSize) + '..';
+  }
+  return input;
+};
+
+//##############################################################################
+promiseList.push((async () => {
+  const mod = await nexusTweaks.registerModule(
+      'characterListTweaks',
+      'Character List Tweaks',
+      'local',
+      'Tweaks for the character display area',
+  );
+
+  const calculateSortValue = (person) => {
+    // higher value = higher up in the list
+    var sortValue = 0;
+
+    // layer 1: allegiance, hostiles sort higher
+    switch (person.politics) {
+      case "faction": sortValue = 0; break;
+      case "ally": sortValue = 10000; break;
+      case "friendly": sortValue = 20000; break;
+      case "neutral": sortValue = 30000; break;
+      case "hostile": sortValue = 40000; break;
+      case "enemy": sortValue = 50000; break;
+    }
+
+    // layer 2: healthiness, hurt people sort higher
+    sortValue += person.currentHp * 1000.0 / person.maxHp;
+
+    return sortValue;
+  }
+
+  const checkStatus = (status, lookFor, abbrev, cssClass = 'status-tag-standard') => {
+    if (status.title.toLowerCase().indexOf(lookFor.toLowerCase()) > -1) {
+      status.abbrev = abbrev;
+      status.class = cssClass;
+    }
+  }
+
+  const createPerson = (charHtml) => {
+    // creates an object of the character's stats from an html string
+    var temp;
+    var person = {
+      'name': null,
+      'politics': null,
+      'level': null,
+      'rank' : null,
+      'hpVisible': false,
+      'currentHp': null,
+      'maxHp': null,
+      'mpVisible': false,
+      'currentMp': null,
+      'maxMp': null,
+      'morality': null,
+      'smDuration': null,
+      'hpTooltip': null,
+      'sortValue': null,
+      'statusTags': []
+    };
+
+    // politics
+    temp = /a[^>]+class="(faction|ally|friendly|neutral|enemy|hostile)"/.exec(
+        charHtml);
+    if (temp) {
+      person.politics = temp[1];
+    }
+
+    // character ID and name
+    temp = /href="javascript:SelectItem\('target_id','(\d+)'\)">(.*?)</.exec(charHtml)
+    if (temp) {
+      person.id = temp[1];
+      person.name = temp[2];
+    }
+
+    // character link (level, and alternate ID)
+    temp = /\(<a href="clash.php\?op=character&amp;id=(\d+)">(\d+)(.?)<\/a>\)/.exec(
```

Comment:
nit: We can be more narrow about ranks, e.g. `[A-Z]?` or even `[A-I]` or whatever the highest rank currently is.